### PR TITLE
Migrate esp-hal dependency to version 0.17.0

### PR DESCRIPTION
**Pull request by 1Kill2Steal**

### CODE REVIEW

**Comment by 1Kill2Steal** [comment]
Hmm... CI fails when the version gets bumped...

**Comment by 1Kill2Steal** [comment]
❯ cargo check --features=esp32c6


WARNING: use --release
  We *strongly* recommend using release profile when building esp-hal.
  The dev profile can potentially be one or more orders of magnitude
  slower than release, and may cause issues with timing-senstive
  peripherals and/or devices.

    Checking esp-ieee802154 v0.1.0 (/home/hutao/Documents/Code/arduino/esp32-c6-dev-kit-n8/esp-ieee802154/esp-ieee802154)
error[E0432]: unresolved import `esp_hal::prelude::interrupt`
  --> src/raw.rs:20:5
   |
20 |     prelude::interrupt,
   |     ^^^^^^^^^^^^^^^^^^ no `interrupt` in `prelude`
   |
   = help: consider importing this module instead:
           esp_hal::interrupt

For more information about this error, try `rustc --explain E0432`.
error: could not compile `esp-ieee802154` (lib) due to 1 previous error


Also it might be worth it to update the CI to use the --release flag. I can work on fixing the module import on this PR if you want.

**Comment by 1Kill2Steal** [comment]
> 0.17 contains breaking changes see https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md#changed-1, would you mind updating the code to use the new changes too?

I don't mind updating it

**Comment by MabezDev** [comment]
I forgot to mention we also have a migration guide which could be helpful: https://github.com/esp-rs/esp-hal/releases/tag/v0.17.0

File: esp-ieee802154/src/raw.rs [hunk 1]
```diff
@@ -143,7 +142,6 @@ fn ieee802154_mac_init() {
     // memset(s_rx_frame, 0, sizeof(s_rx_frame));
     // s_ieee802154_state = IEEE802154_STATE_IDLE;
 
-    interrupt::enable(Interrupt::ZB_MAC, Priority::Priority1).unwrap();
     unsafe {
         esp_hal::riscv::interrupt::enable();
     }
```
Comment (bjoernQ) [inline esp-ieee802154/src/raw.rs:147]: Handlers are not automatically registered anymore in esp-hal 0.17 so we'd need this here:
```rust
    unsafe {
        esp_hal::interrupt::bind_interrupt(
            esp_hal::peripherals::Interrupt::ZB_MAC,
            ZB_MAC.handler(),
        );
    }
    esp_hal::interrupt::enable(esp_hal::peripherals::Interrupt::ZB_MAC, ZB_MAC.priority()).unwrap();
```

**Comment by bjoernQ** [comment]
Thanks for working on this!

We would need to manually bind the interrupt handler since esp-hal 0.17.0

We could get away without depending on `embedded-hal` by using the inherent function `delay_millis` instead of `delay_ms` in the examples

**Comment by bjoernQ** [comment]
Regarding the warnings: From a brief look its mostly unused functions and constants. t.b.h. I'm not sure if we should just remove them - they are basically left-overs from porting the driver written in C (and that's why the code still looks a lot like "C in Rust syntax" in many places). 

I think most of that is for timed send/receive operations which we currently don't support here. Once we implement that we can see that code in git-history anyways

@MabezDev what do you think? having unused code might be confusing and distracting when trying to understand things here

**Comment by 1Kill2Steal** [comment]
I'm leaving the PR as this, if there are any more changes needed then I'll be free to make them after Easter (so after the 6th of May) since I won't be able to be on the PC until then.

Feel free to do any edits if needed or tell me to put the PR as a draft until I'm free to continue working on this.

I wish you both have good holidays (or a nice time if you don't celebrate them)

**Review by bjoernQ — APPROVED** [review]
Looks good to me and works fine, thanks!

**Comment by bjoernQ** [comment]
> I wish you both have good holidays (or a nice time if you don't celebrate them)

Enjoy your holidays

**Pull request merged**